### ISSUE TITLE
[Push Publishing] : The "Push Publishing" Actionlet is throwing a NPE from Tasks portlet

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#17026]
Reported via Support: [Zendesk-98889](https://dotcms.zendesk.com/agent/tickets/98889)
Reproducible in dotCMS 4.3.3, 5.1.6, and latest master.

### Describe the bug
If you go to the **Tasks** portlet, check one or more elements in your task list, select the _Push Publish_ action from the **Workflows** dropdown, and click **Execute Workflow**, you'll get message in the UI stating: `FAILURE: Cannot find execute task null`, and a Java stack trace in the log file, related to a NPE:
```
12:19:04.401  WARN  ajax.WfTaskAjax - cannot find task 6ee16074-edfb-45ff-b157-90f199170bfc :null
com.dotmarketing.portlets.workflows.business.DotWorkflowException: null
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireWorkflowPostCheckin_aroundBody148(WorkflowAPIImpl.java:2024) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl$AjcClosure149.run(WorkflowAPIImpl.java:1) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:40) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireWorkflowPostCheckin(WorkflowAPIImpl.java:1979) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireContentWorkflow_aroundBody162(WorkflowAPIImpl.java:2697) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl$AjcClosure163.run(WorkflowAPIImpl.java:1) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:40) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireContentWorkflow(WorkflowAPIImpl.java:2666) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.portlets.workflows.ajax.WfTaskAjax.executeActions(WfTaskAjax.java:144) ~[dotcms_5.1.6_daab7f6.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_162]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_162]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_162]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_162]
	at com.dotmarketing.portlets.workflows.ajax.WfBaseAction.service(WfBaseAction.java:43) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.portlets.workflows.ajax.WfTaskAjax.service(WfTaskAjax.java:1) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.servlets.ajax.AjaxAction.doPost(AjaxAction.java:66) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.servlets.ajax.AjaxDirectorServlet.service(AjaxDirectorServlet.java:74) ~[dotcms_5.1.6_daab7f6.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) ~[tomcat-websocket.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.CMSFilter.doFilterInternal(CMSFilter.java:199) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:56) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.cms.urlmap.filters.URLMapFilter.doFilter(URLMapFilter.java:121) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.VanityURLFilter.doFilter(VanityURLFilter.java:104) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:399) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.TimeMachineFilter.doFilter(TimeMachineFilter.java:134) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.ThreadNameFilter.doFilter(ThreadNameFilter.java:88) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.CookiesFilter.doFilter(CookiesFilter.java:46) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotmarketing.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:100) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_5.1.6_daab7f6.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) ~[catalina.jar:8.5.32]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140) ~[catalina.jar:8.5.32]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) ~[catalina.jar:8.5.32]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:650) ~[catalina.jar:8.5.32]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87) ~[catalina.jar:8.5.32]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342) ~[catalina.jar:8.5.32]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:800) ~[tomcat-coyote.jar:8.5.32]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66) ~[tomcat-coyote.jar:8.5.32]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:800) ~[tomcat-coyote.jar:8.5.32]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1471) ~[tomcat-coyote.jar:8.5.32]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat-coyote.jar:8.5.32]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_162]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_162]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-util.jar:8.5.32]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_162]
Caused by: com.dotmarketing.portlets.workflows.model.WorkflowActionFailureException
	at com.dotmarketing.portlets.workflows.actionlet.PushPublishActionlet.executeAction(PushPublishActionlet.java:77) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireWorkflowPostCheckin_aroundBody148(WorkflowAPIImpl.java:1999) ~[dotcms_5.1.6_daab7f6.jar:?]
	... 81 more
Caused by: java.lang.NullPointerException
	at com.dotmarketing.portlets.workflows.actionlet.PushPublishActionlet.getEnvironmentsToSendTo(PushPublishActionlet.java:181) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.portlets.workflows.actionlet.PushPublishActionlet.doPushPublish(PushPublishActionlet.java:146) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.portlets.workflows.actionlet.PushPublishActionlet.executeAction(PushPublishActionlet.java:74) ~[dotcms_5.1.6_daab7f6.jar:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireWorkflowPostCheckin_aroundBody148(WorkflowAPIImpl.java:1999) ~[dotcms_5.1.6_daab7f6.jar:?]
	... 81 more
```

### Steps to reproduce the behavior:
1. Go to **Types & Tags > Workflows > System Workflow > In the Unpublished step, click ADD**
2. Create a new action called "Push Publish", and add the **Push Publish** sub-action.
3. Go to the **Home > Tasks** portlet, check the "landing page" Contentlet, select the "Push Publishing" action, and click **Execute Workflow**.
4. You'll see the `FAILURE: Cannot find execute task null` error message, and the stack trace in the log file.

### Expected behavior
The Push Publishing modal should be displayed -- or as mandated by the R&D Team.

## Acceptance Criteria

1. Display the Push Publishing modal.
2. Work in all the supported browsers (don't forget IE11 and iPad)
3. Unit test

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/17471

**Comment by bryanboza** [comment]
We need some work here after adding the action to the workflow, we are unable to execute this and also we have an error on console and the push publish window is nos appearing
SC: https://www.screencast.com/t/jORNjkfjkIi

**Comment by bryanboza** [comment]
The main problems were fixed here, new enhancement created on #17728... Passed QA
